ffi/native/src/lib.rs

```diff
@@ -3,50 +3,47 @@ extern crate neon;
 extern crate ratel;
 
 use neon::vm::{Call, JsResult};
-use neon::js::{JsString, Variant};
+use neon::js::{JsString, JsBoolean};
+use neon::js::error::{JsError, Kind};
 use ratel::{parser, transformer, codegen};
 
 fn transform(call: Call) -> JsResult<JsString> {
     let scope = call.scope;
 
-    let source = match call.arguments.get(scope, 0).unwrap().variant() {
-        Variant::String(handle) => handle.value() + "\n",
-        _                       => panic!("First argument must be a string"),
-    };
+    if call.arguments.len() == 0 {
+        return JsError::throw(Kind::TypeError, "First argument must be a string")
+    }
 
-    let minify = match call.arguments.get(scope, 1).unwrap().variant() {
-        Variant::Boolean(handle) => handle.value(),
-        _                        => false,
-    };
+    let source = try!(try!(call.arguments.require(scope, 0)).check::<JsString>());
+    let minify = try!(try!(call.arguments.require(scope, 1)).check::<JsBoolean>());
```

Comment:
👍 

FYI: On Rust nightly there is a `?` operator which replaces `try!` macro (with some extra whistles that are still a matter for a discussion in a RFC). Basically `try!(foo)` will become `foo?`, so in this case you'd have:

```rust
let source = call.arguments.require(scope, 0)?.check::<JsString>()?;
let minify = call.arguments.require(scope, 1)?.check::<JsBoolean>()?;
```

Gotta stick to `try!` on stable channels for now, but future is wow.